core/src/androidMain/kotlin/Scanner.kt

```diff
@@ -95,5 +103,11 @@ public class AndroidScanner internal constructor(
                 logger.warn(e) { message = "Failed to stop scan. " }
             }
         }
+    }.filter { advertisement ->
+        // Short-circuit (i.e. don't filter) if no `Filter.NamePrefix` filters were provided.
+        if (namePrefixFilters.isEmpty()) return@filter true
+
+        // Perform `Filter.NamePrefix` filtering here, since it isn't supported natively.
+        namePrefixFilters.any { filter -> filter.matches(advertisement.name) }
```

Comment:
thought this was a bug until I realized you have `.startsWith` further down

Reply:
I guess I was shooting for a common `interface` feel for `Filter`, whereas they all internally could have a `matches` function (but with different parameter types — which makes the `interface` infeasible). 🤷